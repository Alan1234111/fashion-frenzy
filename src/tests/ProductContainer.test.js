import React from "react";
import {render} from "@testing-library/react";
import {BrowserRouter as Router} from "react-router-dom";
import ProductContainer from "../components/ProductContainer";

test("renders ProductContainer correctly", () => {
  const props = {
    id: "123",
    image: "image-url",
    title: "Product Title",
    price: 9.99,
    rate: 4.5,
    count: 10,
    handleClick: jest.fn(),
  };
  const {container} = render(
    <Router>
      <ProductContainer {...props} />
    </Router>
  );
  expect(container).toMatchSnapshot();
});
