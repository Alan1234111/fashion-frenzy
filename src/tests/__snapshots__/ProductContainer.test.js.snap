// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProductContainer correctly 1`] = `
<div>
  <a
    class="text-center p-5 bg-white shadow-sm text-decoration-none text-black col-md-3"
    href="/123"
  >
    <div
      class="d-flex justify-content-center align-items-center h-50"
    >
      <img
        alt=""
        class="mb-4"
        src="image-url"
        style="width: 150px;"
      />
    </div>
    <h3
      class="fs-6 product-title"
    >
      Product Title
    </h3>
    <h4
      class="fs-5 mt-4 fw-bold"
    >
      $
      9.99
    </h4>
    <div
      class="rating mb-3"
    >
      <img
        alt=""
        class="star-fill"
        src="star-fill.svg"
      />
      <img
        alt=""
        class="star-fill"
        src="star-fill.svg"
      />
      <img
        alt=""
        class="star-fill"
        src="star-fill.svg"
      />
      <img
        alt=""
        class="star-fill"
        src="star-fill.svg"
      />
      <img
        alt=""
        class="star-fill"
        src="star-fill.svg"
      />
      <span
        class="ms-1"
      >
        10
      </span>
    </div>
    <button
      class="mt-3 index-1 btn btn-danger"
      type="button"
    >
      See More
    </button>
  </a>
</div>
`;
